### PR TITLE
ci: setup python workflow

### PR DESCRIPTION
**Pull request by universalmind303**

### CODE REVIEW

File: .github/workflows/python-release.yaml [hunk 1]
```diff
@@ -0,0 +1,140 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - cory/py-ci
```
Comment (universalmind303) [inline .github/workflows/python-release.yaml:6]: revert before merging

File: .github/workflows/python-release.yaml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+on:
+  push:
+    tags:
+      - "py-*"
+  workflow_dispatch:
+
+permissions:
+  contents: read
+env:
+  RUST_TOOLCHAIN: nightly-2023-06-23
```
Comment (scsmithr) [inline .github/workflows/python-release.yaml:12]: Do we need nightly?

File: .github/workflows/python-release.yaml [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+          manylinux: auto
+          container: 'off'
+      - name: Upload wheels
+        uses: actions/upload-artifact@v3
+        with:
+          name: wheels
+          path: py-glaredb/dist
+
+  windows:
+    runs-on: windows-latest
```
Comment (scsmithr) [inline .github/workflows/python-release.yaml:49]:
```suggestion
    runs-on: windows-latest-8-cores
```

File: .github/workflows/python-release.yaml [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+env:
+  RUST_TOOLCHAIN: nightly-2023-06-23
+  PYTHON_VERSION: '3.7'
+  MATURIN_VERSION: '1.0.1'
+  MATURIN_PYPI_TOKEN: ${{ secrets.PYPI_API_TOKEN }}
+  MATURIN_USERNAME: 'seanglaredb' 
+  PROTOC: "${{ github.workspace }}/deps/protoc/bin/protoc"
+jobs:
+  linux:
+    runs-on: ubuntu-latest
```
Comment (scsmithr) [inline .github/workflows/python-release.yaml:20]:
```suggestion
    runs-on: ubuntu-2004-8-cores
```

File: .github/workflows/python-release.yaml [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+          sccache: 'true'
+          container: 'off'
+      - name: Upload wheels
+        uses: actions/upload-artifact@v3
+        with:
+          name: wheels
+          path: py-glaredb/dist
+
+  macos:
+    runs-on: macos-latest
```
Comment (scsmithr) [inline .github/workflows/python-release.yaml:80]:
```suggestion
    runs-on: macos-12-xl
```

**Review by scsmithr — APPROVED** [review]
Minor comments, runners should be drop-in replacements.

**Pull request merged**